kyo-data/shared/src/main/scala/kyo/Schedule.scala

```diff
@@ -0,0 +1,348 @@
+package kyo
+
+import Schedule.internal.*
+import kyo.Duration
+import kyo.Instant
+
+/** An immutable, composable scheduling policy.
+  *
+  * Schedule provides various combinators for creating complex scheduling policies. It can be used to define retry policies, periodic tasks,
+  * or any other time-based scheduling logic.
+  */
+sealed abstract class Schedule derives CanEqual:
+
+    /** Returns the next delay and the updated schedule.
+      *
+      * @return
+      *   a tuple containing the next delay duration and the updated schedule
+      */
+    def next: Maybe[(Duration, Schedule)]
```

Comment:
I am not in love with this signature. In general, I am not a fan of Tuples...

Reply:
Yeah, I had a `withNext` instead in one of the impls I pushed because of that. I decided to go back to the tuple because this is an API I'd expect users to rarely use directly and the tuple simplifies the implementation.